cmd/thanos/sidecar.go

```diff
@@ -153,21 +154,20 @@ func runSidecar(
 	})
 
 	// Setup all the concurrent groups.
+	promAgentModeEnabled := false
 	{
 		promUp := promauto.With(reg).NewGauge(prometheus.GaugeOpts{
 			Name: "thanos_sidecar_prometheus_up",
 			Help: "Boolean indicator whether the sidecar can reach its Prometheus peer.",
 		})
-
 		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		// Check prometheus's flags to ensure same sidecar flags.
+		if err := validatePrometheus(ctx, m.client, logger, conf.shipper.ignoreBlockSize, m, uploads, &promAgentModeEnabled); err != nil {
```

Comment:
Why not include this in `g.Add()`?

Reply:
What was happening was that the promAgentModeEnabled variable was not being evaluated and so the code block from line 303 was picking it as `false` when it was in agent mode